lib/dgram.js

```diff
@@ -266,10 +267,12 @@ Socket.prototype.sendto = function(buffer,
 
 
 function sliceBuffer(buffer, offset, length) {
-  if (typeof buffer === 'string')
+  if (typeof buffer === 'string') {
     buffer = Buffer.from(buffer);
-  else if (!(buffer instanceof Buffer))
-    throw new TypeError('First argument must be a buffer or string');
+  } else if (!isUint8Array(buffer)) {
+    throw new TypeError('First argument must be a buffer, ' +
```

Comment:
Since the error message is changing anyway, can you s/buffer/Buffer here to make it consistent with the other error messages thrown in this module?